### PR TITLE
Feature: VRF random number

### LINKED ISSUE #3414: Add VRF random number
Add VRF random number implementation to AEDPoS.

### PR DESCRIPTION
**Pull request by zhxymh**
Closes #3414

### CODE REVIEW

File: src/AElf.ContractTestKit.AEDPoSExtension/BlockMiningService.cs [hunk 1]
```diff
@@ -156,6 +161,15 @@ public async Task MineBlockAsync(List<Transaction> transactions = null, bool wit
 
         _isSkipped = false;
     }
+    
+    private async Task<byte[]> GenerateRandomProveAsync()
```
Comment (gldeng) [inline src/AElf.ContractTestKit.AEDPoSExtension/BlockMiningService.cs:165]: Should be `GenerateRandomProofAsync`.

File: src/AElf.Kernel.Consensus.AEDPoS/Application/IRandomProvider.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+using System.Threading.Tasks;
+using AElf.Kernel.Account.Application;
+using AElf.Types;
+using Volo.Abp.DependencyInjection;
+
+namespace AElf.Kernel.Consensus.AEDPoS.Application;
+
+public interface IRandomProvider
```
Comment (gldeng) [inline src/AElf.Kernel.Consensus.AEDPoS/Application/IRandomProvider.cs:8]: Will `IRandomNumberProvider` be more accurate?

File: src/AElf.Kernel.Consensus.AEDPoS/Application/IAEDPoSInformationProvider.cs [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
+using AElf.Types;
 
 namespace AElf.Kernel.Consensus.AEDPoS.Application;
 
 // ReSharper disable once InconsistentNaming
 public interface IAEDPoSInformationProvider
 {
-    Task<IEnumerable<string>> GetCurrentMinerList(ChainContext chainContext);
+    Task<IEnumerable<string>> GetCurrentMinerListAsync(ChainContext chainContext);
+    Task<Hash> GetRandomHashAsync(IChainContext chainContext, long blockHeight);
```
Comment (gldeng) [inline src/AElf.Kernel.Consensus.AEDPoS/Application/IAEDPoSInformationProvider.cs:11]: Better separate this method in another interface.

File: src/AElf.Kernel.SmartContract/HostSmartContractBridgeContext.cs [hunk 4]
```diff
@@ -349,4 +349,9 @@ public byte[] RecoverPublicKey(byte[] signature, byte[] hash)
         var cabBeRecovered = CryptoHelper.RecoverPublicKey(signature, hash, out var publicKey);
         return !cabBeRecovered ? null : publicKey;
     }
+
+    public byte[] ECVrfVerify(byte[] pubKey, byte[] alpha, byte[] pi)
+    {
+        return CryptoHelper.ECVrfVerify(pubKey, alpha, pi);
```
Comment (gldeng) [inline src/AElf.Kernel.SmartContract/HostSmartContractBridgeContext.cs:355]: This method should not throw Exception.

**Review by gldeng — APPROVED** [review]

**Review by jason-hoopox — APPROVED** [review]

**Pull request merged**